bucket/qq.json

```diff
@@ -0,0 +1,41 @@
+{
+    "version": "9.7.3.28946",
+    "description": "An instant messaging tool that gives you the best way to keep in touch with your friends and family.",
+    "homepage": "https://im.qq.com",
+    "license": {
+        "identifier": "Proprietary",
+        "url": "https://ti.qq.com/agreement/index.html"
+    },
+    "url": "https://webcdn.m.qq.com/spcmgr/download/QQ9.7.3.28946.exe#/dl.7z",
+    "hash": "6d11c322a7deb69634de54ba3006f908fa53a09a63585aaf40b02d3dce6fd825",
+    "shortcuts": [
+        [
+            "Bin\\QQScLauncher.exe",
+            "QQ"
+        ]
+    ],
+    "installer": {
+        "script": [
+            "Get-Item \"$dir\\*\" -Exclude .rsrc | Remove-Item",
+            "Expand-7zipArchive \"$dir\\.rsrc\\2052\\MSI\\101\" \"Files\" -Destination \"$dir\\\" -Removal",
+            "Move-Item \"$dir\\Files\\*\" \"$dir\"",
+            "Expand-7zipArchive \"$dir\\.rsrc\\2052\\MSI\\102\" \"vcruntime140.dll\" -Destination \"$dir\\Bin\\\"",
+            "Expand-7zipArchive \"$dir\\.rsrc\\2052\\MSI\\102\" \"msvcp140.dll\" -Destination \"$dir\\Bin\\\"",
+            "Expand-7zipArchive \"$dir\\.rsrc\\2052\\MSI\\102\" \"msvcr100.dll\" -Destination \"$dir\\Bin\\\"",
+            "Expand-7zipArchive \"$dir\\.rsrc\\2052\\MSI\\102\" \"vcomp140.dll\" -Destination \"$dir\\Bin\\\"",
+            "Expand-7zipArchive \"$dir\\.rsrc\\2052\\MSI\\102\" \"msvcp100.dll\" -Destination \"$dir\\Bin\\\"",
+            "Expand-7zipArchive \"$dir\\.rsrc\\2052\\MSI\\102\" \"TXSSO\\TXSSOSetup.exe\" -Destination \"$dir\" -Removal",
+            "Expand-7zipArchive \"$dir\\TXSSO\\TXSSOSetup.exe\" \"TXSSO\" -Destination \"$dir\\bin\" -Removal"        ]
+    },
+    "post_install": "Echo \"$dir\\Files\" \"$dir\\.rsrc\" \"$dir\\TXSSO\" | Remove-Item -Recurse",
+    "uninstaller": {
+        "script": "Remove-Item \"$env:APPDATA\\Tencent\\QQ\" -Recurse -Force"
```

Comment:
This is not a good practice as it'll erase the runtime-generated data every time an upgrade is triggered. And these data will be re-generated when the app starts. Hence it makes no sense. App data should only be manipulated while the `--purge` flag is used.

Reply:
I have modified it according to your comment and added some persist configuration, please check it out and feel free to review and comment.